OnMessageTask.h

```diff
@@ -31,75 +31,82 @@ class OnMessageTask : public tbb::task {
 };
 
 class MultiMessageTask : public OnMessageTask {
-  std::tuple<uint8_t,uint32_t,std::vector<char>> _msg;
-  tbb::task* execute();
+    CARTA::EventHeader header;
+    int event_length;
+    char * event_buffer;
+    tbb::task* execute();
 public:
- MultiMessageTask(Session *session_,
-		  std::tuple<uint8_t,uint32_t,std::vector<char>> msg)
-   : OnMessageTask( session_ ) {
-    _msg= msg;
+ MultiMessageTask(Session *session, CARTA::EventHeader& head,
+		  int evt_len, char* event_buf)
+   : OnMessageTask( session ) {
+      header= head;
+      event_length= evt_len;
+      event_buffer= event_buf;
   }
   ~MultiMessageTask() {}
 };
 
 class SetImageChannelsTask : public OnMessageTask {
-  std::pair<CARTA::SetImageChannels,uint32_t> _req;
+  std::pair<CARTA::SetImageChannels,uint32_t> req;
   tbb::task* execute();
 public:
- SetImageChannelsTask(Session *session_,
-		      std::pair<CARTA::SetImageChannels,uint32_t> req_ )
-       : OnMessageTask( session_ ) {
-    _req= req_;
+ SetImageChannelsTask(Session *session,
+		      std::pair<CARTA::SetImageChannels,uint32_t> req )
+       : OnMessageTask( session ) {
+    req= req;
```

Comment:
Missed a space, but we're about to reformat everything anyway so it's not super important.